brave/brave-core
# Optimize resource loading for several profiles.

Resolves https://github.com/brave/brave-browser/issues/30659



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

* Start with a fresh profile, opt-in to Brave Ads
* Check that:
  -   the text classification resource is loaded
  -   text embedding resource is loaded
* Check that:
  -   the visited web pages classification works
  -   text embedding processing works: https://github.com/brave/brave-core/pull/14393

* Create another Browser profile
* Opt-in to Brave Ads in a new profile
* For a new profile check that:
  -   the text classification resource is loaded
  -   text embedding resources are loaded
* For a new profile check that: 
  -   the visited web pages classification works
  -   text embedding processing works: https://github.com/brave/brave-core/pull/14393
